app/containers/HomePage/saga.js

```diff
@@ -20,7 +23,7 @@ export function* getRepos() {
   try {
     // Call our request helper (see 'utils/request')
     const repos = yield call(request, requestURL);
-    yield put(reposLoaded(repos, username));
+    yield put(reposLoaded({ repos, username }));
```

Comment:
👌👌👌

Reply:
This is due to the behavior of `createAction()`.  By default, generated action creators accept a single argument, which becomes `action.payload`.  That does put the responsibility on the dispatching code to format the payload contents correctly.

If a developer wants to do some formatting of the payload contents, such as generating a unique ID or turning multiple action creators args into a payload, they can define a "prepare callback":

https://redux-toolkit.js.org/api/createaction#using-prepare-callbacks-to-customize-action-contents

https://redux-toolkit.js.org/api/createslice#customizing-generated-action-creators